esp/services/ws_dfu/ws_dfuService.hpp

```diff
@@ -95,6 +95,27 @@ class CThorNodeGroupCache: public SuperHashTableOf<CThorNodeGroup, const char>
     CThorNodeGroup *lookup(const char* groupName, unsigned timeOutMinutes);
 };
 
+struct DFUReadAccessRequest
+{
+    const char *logicalName = nullptr;
+    const char *clusterName = nullptr;
+    IUserDescriptor *udesc  = nullptr;
+    CSecAccessType accessType;
+    unsigned expiry = 1;
+    bool refresh = false;
+    bool returnJsonTypeInfo = false;
+    bool returnBinTypeInfo = false;
+};
+
+struct DFUReadAccessResponse
+{
+    unsigned numParts = 0;
+    IArrayOf<IEspDFUPartLocations> dfuPartLocations;
+    IArrayOf<IEspDFUPartCopies> dfuPartCopies;
+    MemoryBuffer binLayout;
+    StringBuffer jsonLayout;
+};
```

Comment:
@wangkx - the above 2 structures are no loner being used and out of date. Have left them in for now, for the 'create and publish' case